src/EventSubscriber/SmsMessageProcessor.php

```diff
@@ -162,9 +162,13 @@ protected function getGatewayForPhoneNumber($recipient) {
   public function deliveryReportUrl(SmsMessageEvent $event) {
     foreach ($event->getMessages() as &$sms_message) {
       if (!$sms_message->getOption('delivery_report_url')) {
-        $route_parameters = ['sms_gateway' => $sms_message->getGateway()->id()];
-        $url = Url::fromRoute('sms.process_delivery_report', $route_parameters);
-        $sms_message->setOption('delivery_report_url', $url->setAbsolute()->toString());
+        $url = $sms_message->getGateway()->getPushReportUrl();
+        try {
+          $url = $url->setAbsolute()->toString();
```

Comment:
>  attaching generated tokens

Yeh this sounds like it would require a bit of investigation. Out of scope for now.